cmd/buf/internal/command/registry/registrylogin/registrylogin.go

```diff
@@ -284,6 +284,10 @@ func doBrowserLogin(
 	if err != nil {
 		return "", err
 	}
+	baseURL := buftransport.PrependHTTPS(remote)
+	if appConfig.TLS == nil {
+		baseURL = buftransport.PrependHTTP(remote)
+	}
```

Comment:
drive-by, but: you want to `if` / `else` this, otherwise we'd double-prepend in the http case, right?